game-core/src/main/java/games/strategy/triplea/ResourceLocationTracker.java

```diff
@@ -10,42 +15,30 @@
  */
 @UtilityClass
 class ResourceLocationTracker {
+  // All maps must have at least a "baseTiles" folder
+  private static final String REQUIRED_ASSET_EXAMPLE_FOLDER = "baseTiles/";
 
   /**
-   * master zip is the zipped folder format you get when downloading from a map repo via the 'clone
-   * or download' button.
-   */
-  static final String MASTER_ZIP_MAGIC_PREFIX = "-master/map/";
-
-  static final String MASTER_ZIP_IDENTIFYING_SUFFIX = "-master.zip";
-
-  /**
-   * * Will return an empty string unless a special prefix is needed, in which case that prefix is *
-   * constructed based on the map name. * *
-   *
-   * <p>The 'mapPrefix' is the path within a map zip file where we will then find any map contents.
-   * * For example, if the map prefix is "map", then when we expand the map zip, we would expect *
-   * "/map" to be the first folder we see, and we would expect things like "/map/game" and *
-   * "/map/polygons.txt" to exist.
+   * Will return an empty string unless a special prefix is needed, in which case that prefix is *
+   * constructed based on where the {@code baseTiles} folder is located within the zip.
    *
-   * @param mapName Used to construct any special resource loading path prefixes, used as needed
-   *     depending upon which resources are in the path
    * @param resourcePaths The list of paths used for a map as resources. From this we can determine
    *     if the map is being loaded from a zip or a directory, and if zip, if it matches any
    *     particular naming.
    */
-  static String getMapPrefix(final String mapName, final URL[] resourcePaths) {
-    final boolean isUsingMasterZip =
-        Arrays.stream(resourcePaths)
-            .map(Object::toString)
-            .anyMatch(path -> path.endsWith(MASTER_ZIP_IDENTIFYING_SUFFIX));
-
-    // map skins will have the full path name as their map name.
-    if (mapName.endsWith("-master.zip")) {
-      return mapName.substring(0, mapName.length() - "-master.zip".length())
-          + MASTER_ZIP_MAGIC_PREFIX;
-    } else {
-      return isUsingMasterZip ? mapName + MASTER_ZIP_MAGIC_PREFIX : "";
+  static String getMapPrefix(final URL[] resourcePaths) {
+    for (final URL url : resourcePaths) {
+      try (ZipFile zip = new ZipFile(new File(url.toURI()))) {
+        final Optional<? extends ZipEntry> e =
+            zip.stream().filter($ -> $.getName().endsWith(REQUIRED_ASSET_EXAMPLE_FOLDER)).findAny();
+        if (e.isPresent()) {
+          final String path = e.get().getName();
+          return path.substring(0, path.length() - REQUIRED_ASSET_EXAMPLE_FOLDER.length());
```

Comment:
I notice we are substringing off only the example folder name. Seemingly we want the path to be something like `<map-name>/map`. If the 'baseTiles` is not in that directory, then we might substring too little. For example, let's say `baseTiles` is in a sub-folder of 'map', then the path would no longer be to '/map' but would be to that subfolder. Wouldn't this cause a problem with files being one-folder level higher than expected?

Reply:
If I understand correctly, this is exactly the problem that this PR is trying to solve. If the map maker for some reason puts the assets in `/map/stuff/baseTiles` this code will use `/map/stuff` as the root path of the resources, not `/map` as was previously hardcoded and assumed. This stops making assumptions about absolute paths for resources and instead makes the assumption that, wherever `baseTiles` is, all the other folders are siblings to that folder.